script.js

```diff
@@ -63,21 +63,39 @@ function renderTodos() {
 
   filteredTodos.forEach((todo, index) => {
     const listItem = document.createElement("li");
+    listItem.style.display = "flex";
+
     const checkbox = document.createElement("input");
     checkbox.type = "checkbox";
     checkbox.checked = todo.checked;
     listItem.appendChild(checkbox);
 
     const titleText = document.createElement("span");
     titleText.textContent = todo.title;
+    titleText.style.flex = "1";
     listItem.appendChild(titleText);
 
+    const deleteButton = document.createElement("img");
+    deleteButton.src =
+      "https://pluspng.com/img-png/delete-button-png-delete-icon-1600.png";
+    deleteButton.alt = "Delete";
+    deleteButton.style.width = "20px";
+    deleteButton.style.height = "20px";
+    listItem.appendChild(deleteButton);
+
     checkbox.addEventListener("change", () => {
       state.todos[index].checked = checkbox.checked;
       updateUncheckedTodoCount();
       sessionStorage.setItem(SESSIONSTORAGE_KEY, JSON.stringify(state.todos));
     });
 
+    deleteButton.addEventListener("click", () => {
+      state.todos.splice(index, 1);
```

Comment:
this is a bit complicated 🤫
despite it works perfectly you'd better keep your state immutable, so you need to produce a new updated array and assign it to the state:
```
const newTodos = [...state.todos];
newTodos.splice(index, 1);
state.todos = newTodos;
```

Reply:
Could you explain immutability of the state in a bit more detail?

Reply:
Okay you have already noticed processes in a browser are of asynchronous nature haven't you? And global state might be used by several processes simultaneously. If it is mutable then some process might experience discrepancies caused by changes of a state during its run. Imagine a large scale app which has an array of something as a state, okay? And on of its processes is doing heavy calculations on the state. And during that time consuming process a user clicks a button which deletes an element of the state array. If it's not immutable those calculations may fail and error out. If it's immutable then clicking of delete button would crest a new state array while the process of heavy calculations would be still dealing with the previous version of the state array. Not sure if I'm clear enough here, please ask me on an occasion for more details okay?

Reply:
ChatGPT suggests creating two subarrays: state.todos = [...state.todos.slice(0, index), ...state.todos.slice(index + 1)];

Reply:
`state.todos = [...state.todos].splice(index, 1);` 

This seems to have the opposite effect and delete all the todos except the one that is clicked.

Reply:
😀😀😀
sure `splice` returns deleted items. Should be:
```
const newTodos = [...state.todos];
newTodos.splice(index, 1);
state.todos = newTodos;
```